internal/tsoptions/parsedcommandline.go

```diff
@@ -2,20 +2,52 @@ package tsoptions
 
 import (
 	"slices"
+	"sync"
 
 	"github.com/microsoft/typescript-go/internal/ast"
 	"github.com/microsoft/typescript-go/internal/core"
+	"github.com/microsoft/typescript-go/internal/tspath"
+	"github.com/microsoft/typescript-go/internal/vfs"
 )
 
 type ParsedCommandLine struct {
 	ParsedConfig *core.ParsedOptions `json:"parsedConfig"`
 
-	ConfigFile *TsConfigSourceFile `json:"configFile"` // TsConfigSourceFile, used in Program and ExecuteCommandLine
-	Errors     []*ast.Diagnostic   `json:"errors"`
-	Raw        any                 `json:"raw"`
-	// WildcardDirectories map[string]watchDirectoryFlags
-	CompileOnSave *bool `json:"compileOnSave"`
+	ConfigFile    *TsConfigSourceFile `json:"configFile"` // TsConfigSourceFile, used in Program and ExecuteCommandLine
+	Errors        []*ast.Diagnostic   `json:"errors"`
+	Raw           any                 `json:"raw"`
+	CompileOnSave *bool               `json:"compileOnSave"`
 	// TypeAquisition *core.TypeAcquisition
+
+	comparePathsOptions     tspath.ComparePathsOptions
+	wildcardDirectoriesOnce sync.Once
+	wildcardDirectories     map[string]bool
+	extraFileExtensions     []fileExtensionInfo
+}
+
+// WildcardDirectories returns the cached wildcard directories, initializing them if needed
+func (p *ParsedCommandLine) WildcardDirectories() map[string]bool {
+	if p.wildcardDirectories != nil {
+		return p.wildcardDirectories
+	}
```

Comment:
You could actually eliminate this branch entirely, as the Do call is already a branch that checks that this has been set.

Reply:
Oh, I see, funky

Reply:
Open to other suggestions. Is there a safe way to copy the Once state?

Reply:
There isn't, no; if it can be accessed atomically then it can't be traditionally copied by value.